src/layers.rs

```diff
@@ -1,126 +1,112 @@
-                    for tile_y in 0..target_chunk_y {
-                        let lookup_x = (chunk_x * target_chunk_x) + tile_x;
-                        let lookup_y = (chunk_y * target_chunk_y) + tile_y;
-                        let chunk_pos = Vec2::new(lookup_x as f32, lookup_y as f32);
+        let _tile_space = tileset.spacing as f32; // TODO: re-add tile spacing.. :p
 
-                        tiles_y.push(
-                            if lookup_x < map.width as usize && lookup_y < map.height as usize {
-                                let map_tile = match &layer.tiles {
-                                    tiled::LayerData::Finite(tiles) => &tiles[lookup_y][lookup_x],
-                                    _ => panic!("Infinte maps not supported"),
-                                };
-                                // tile not in this set
-                                if map_tile.gid < tileset.first_gid
-                                    || map_tile.gid
-                                        >= tileset.first_gid + tileset.tilecount.unwrap()
-                                {
-                                    continue;
-                                }
-                                // Calculate positions
-                                let vertex = match map.orientation {
-                                    tiled::Orientation::Orthogonal => {
-                                        let center =
-                                            project_ortho(chunk_pos, tile_width, tile_height);
+        let mut map = Map::new(
+            Vec2::new((tiled_map.width as f32 / 64.0).ceil(), (tiled_map.height as f32 / 64.0).ceil()).into(), 
+            Vec2::new(64.0, 64.0).into(),
+            Vec2::new(tile_width, tile_height),
+            Vec2::new(tileset.images[0].width as f32, tileset.images[0].height as f32), // TODO: support multiple tileset images?
+            layer.layer_index,
+        );
+        map.mesher = match tiled_map.orientation {
+            tiled::Orientation::Hexagonal => {
+              Box::new(HexChunkMesher::new(HexType::ColumnEven))
+            },
+            tiled::Orientation::Isometric => {
+                Box::new(IsoChunkMesher)
+            },
+            tiled::Orientation::Orthogonal => {
+                Box::new(SquareChunkMesher)
+            },
+            _ => panic!("Unknown tile map orientation!")
+        };
 
-                                        let start = Vec2::new(
-                                            center.x,
-                                            center.y - tile_height - tile_space,
-                                        );
+        // Create layer map rendering entity as child of the tiled map.
+        let mut map_entity = None;
+        commands.entity(parent_entity).with_children(|child_builder| {
+            map_entity = Some(child_builder.spawn().id());
+        });
+        let map_entity = map_entity.unwrap();
```

Comment:
One thing we may want to consider down the line is how we would spawn multiple versions of the same tile map layer. Autoreload is super helpful during development, but we were limited in improving this API by the fact that there was no top-level entity. I like the potential we've added with the new structure.

Also, this construction with the unwrap() seems odd to me, though I see why you do it.  If this is the only option, maybe we can change to expect() so that we can easily see what went wrong if there's a panic?

Reply:
In bevy_ecs_tilemap I would like to have the current `Map` entity be a layer and then create a new Map entity which has layer children. The hierarchy would look like:
`Map > Layer > Chunk > Tile`